pandas/core/groupby/ops.py

```diff
@@ -837,11 +838,22 @@ def names(self):
 
     @property
     def groupings(self) -> "List[grouper.Grouping]":
+        codes, _, _ = self.group_info
+
+        if self.indexer is not None and len(self.indexer) != len(codes):
+            groupers = self.levels
+        else:
+            groupers = [self.result_index._constructor(
+                Categorical.from_codes(self.codes_info, self.result_index))]
+
         return [
             grouper.Grouping(lvl, lvl, in_axis=False, level=None, name=name)
-            for lvl, name in zip(self.levels, self.names)
+            for lvl, name in zip(groupers, self.names)
         ]
 
+    def __iter__(self):
```

Comment:
the issue then len(Grouper) != ngroups right? why are you adding ``__iter__``? rather than for example overriding ``__len__``?

Reply:
Indeed changing `__iter__` was a bad idea since this changes the behavior of resample. Anyway, iterating over a nested `BinGrouper` is somewhat part of the problem (check https://github.com/pandas-dev/pandas/issues/33132#issuecomment-605712931 plz). #33303 probably has a better approach to resolve this.